code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -24,6 +28,192 @@
+			if(fuselength in list(1, 2))
+				user.visible_message(SPAN_DANGER("<b>[user]</b> tries to light the cable on \the [name] but it was too short!"), SPAN_DANGER("You try to light the fuse but it was too short!"))
+				detonate(TRUE) // if you're somehow THAT determined and/or ignorant you managed to get the fuse below 3 seconds, so be it. reap what you sow.
+			else
+				user.visible_message(SPAN_WARNING("<b>[user]</b> lights the cable on \the [name] with \the [W]!"), SPAN_WARNING("You light the cable on \the [name] with the [W]!"))
+				detonate(FALSE)
+	. = ..()
+
+/obj/item/reagent_containers/food/drinks/cans/proc/detonate(var/instant)
+	var/fuel = reagents.get_reagent_amount(/datum/reagent/fuel)
+	if(instant)
+		fuselength = 0
+	else if(prob(fuselength * 6)) // the longer the fuse, the higher chance it will fizzle out (18% chance minimum)
+		var/fizzle = rand(1, fuselength - 1)
+		sleep(fizzle * 10)
+		
+		fuselength -= fizzle
+		visible_message(SPAN_WARNING("The cable on \the [name] fizzles out early."))
+		fuselit = FALSE
+		update_icon()
+		return
+	sleep(fuselength * 10)
+
+	switch(fuel)
+		if(0)
+			visible_message(SPAN_NOTICE("\The [name]'s cable burns out and nothing happens."))
+			fuselit = FALSE
+			fuselength = 0
+			update_icon()
+		if(1 to 10) // baby explosion
+			var/obj/item/trash/can/popped_can = new /obj/item/trash/can(get_turf(src))
+			popped_can.icon_state = icon_state
+			popped_can.name = "popped can"
+			playsound(get_turf(src), 'sound/effects/snap.ogg', 50)
+			visible_message(SPAN_WARNING("\The [name] pops harmlessly!"))
+			fuselit = FALSE // shouldn't have to be done since it's going to get qdel'd, but in case something wacky happens in live play.
+			qdel(src)
+		if(11 to 20) // slightly less baby explosion
+			new /obj/item/material/shard/shrapnel(get_turf(src))
+			playsound(get_turf(src), 'sound/effects/bang.ogg', 50)
+			visible_message(SPAN_WARNING("\The [name] bursts violently into pieces!"))
+			fuselit = FALSE
+			qdel(src)
+		if(21 to INFINITY) // boom
+			fragem(src, 7, 7, 1, 0, 5, 1, TRUE, 2) // The main aim of the grenade should be to hit and wound people with shrapnel instead of causing a lot of station damage, hence the small explosion radius
+			playsound(get_turf(src), 'sound/effects/Explosion1.ogg', 50)
+			visible_message(SPAN_DANGER("<b>\The [name] explodes!</b>"))
+			fuselit = FALSE
+			qdel(src)
+
+/obj/item/reagent_containers/food/drinks/cans/proc/can_light() // just reverses the fuselit var to return a TRUE or FALSE, should hopefully make things a little easier if someone adds more fuse interactions later.
+	if(fuselit)
+		return FALSE
+	else if(fuselength && !fuselit)
+		return TRUE
+
+/obj/item/reagent_containers/food/drinks/cans/bullet_act(obj/item/projectile/P)
+	visible_message(SPAN_DANGER("\The [name] is hit by the [P]!"))
+	if(P.firer && reagents.get_reagent_amount(/datum/reagent/fuel) >= 21)
```

Comment:
This `21` number is a bit of a magic number, either make it into a define or do something like `if(reagents.get_reagent_amount(/datum/reagent/fuel) >= reagents.max_size * 0.75)`